gateway/src/endpoints/feedback.rs

```diff
@@ -308,6 +326,35 @@ async fn write_boolean(
     Ok(())
 }
 
+// Validates the target id and returns the identifier for the target.
+// The identifier will be function name for inference level metrics.
+async fn get_target_identifier(
+    connection_info: &ClickHouseConnectionInfo,
+    metric_config_level: &MetricConfigLevel,
+    target_id: &Uuid,
+) -> Result<String, Error> {
+    match metric_config_level {
+        MetricConfigLevel::Inference => {
+            let query = format!(
+                "SELECT function_name FROM InferenceById WHERE id = '{}'",
+                target_id
+            );
+            let function_name = connection_info.run_query(query).await?.trim().to_string();
+            if function_name.is_empty() {
+                return Err(Error::new(ErrorDetails::InvalidRequest {
+                    message: format!("Inference ID: {target_id} does not exist"),
+                }))
+                .into();
+            };
+            return Ok(function_name);
+        }
+        MetricConfigLevel::Episode => {
+            // TODO: Implement this later as it might require db migrations.
+            return Ok("".to_string());
+        }
+    }
+}
+
```

Comment:
I think this is a good thing to do (since demonstrations need the function name and every other kind of feedback doesn't care) but we should document it more thoroughly for future developers. 

Can you lay out the rationale more explicitly in the doc comment?